File: cpp/linked-lists/0141-linked-list-cycle.cpp
#include <structures.h>

bool hasCycle(ListNode *head) {
    ListNode* fast = head;
    ListNode* slow = head;
    if (head && head->next) {
        while(fast->next && fast->next->next) {
            fast = fast->next->next;
            slow = slow->next;
            if (fast == slow) return true;
        }
    }
    return false;
}

File: rust/linked-lists/0141-linked-list-cycle.rs
from structures import ListNode

def hasCycle(head: ListNode) -> bool:
    fast = head
    slow = head
    
    if head:
        while fast.next and fast.next.next:
            fast = fast.next.next
            slow = slow.next
            
            if fast == slow:
                return True
        
    return False